src/components/workflow/workflow.service.ts

```diff
@@ -0,0 +1,866 @@
+          ]
+        ])
+        .detachDelete('rel')
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not remove security group from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // changeCurrentStateInWorkflow
+  async changeCurrentState(session: ISession, input: ChangeState): Promise<CommentState>{
+    try{
+
+      await this.db
+        .query()
+        .match([
+
+        ])
+        .run();
+    } catch (e) {
+      this.logger.warning('could not change current state', {
+        exception: e
+      });
+      throw e;
+    }
+    throw new NotImplementedException();
+  }
+
+  // creates a relationship from one state to another
+  // later we can create an abstracted function that creates and attaches a state to another state
+  // addPossibleStateToState
+  async addPossibleState(session: ISession, input: PossibleState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup')
+          ],
+          [
+            node('fromState', 'State', {
+              id: input.fromStateId
+            })
+          ],
+          [
+            node('toState', 'State', {
+              id: input.toStateId
+            })
+          ]
+        ])
+        .merge([
+          node('fromState'),
+          relation('out', '', 'nextPossibleState', {
+            active: true
+          }),
+          node('toState')
+        ])
+        .run();
+    } catch (e) {
+      this.logger.warning('could not add possible state to state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // removePossibleStateFromState
+  async removePossibleState(session: ISession, input: PossibleState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup'),
+            relation('in', '', 'securityGroup', {
+              active: true
+            }),
+            node('fromState', 'State', {
+              id: input.fromStateId
+            }),
+            relation('out', 'rel', 'nextPossibleState', {
+              active: true
+            }),
+            node('toState', 'State', {
+              id: input.toStateId
+            })
+          ],
+        ])
+        .detachDelete('rel')
+        .run();
+
+    } catch (e) {
+      this.logger.warning('failed to remove possible state from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // there will be more than one required field relationship between a state node and a base node.
+  // this is so each required field can be queried without inspecting the property name in app code.
+  // addRequiredFieldToState
+  async addField(session: ISession, input: RequiredField): Promise<void>{
```

Comment:
the name of this relationship is literally: `:requiredField` it has a property named `value` which is set to the identifier of the field that is required, like `name`. So the relationship object would like: `-[:requiredField {value: "name"}]->`

Reply:
should change the name to addRequiredField